java/Anton-Kozyk/src/com/zoolatech/lecture5/tasks/_4/Task4.java

```diff
@@ -0,0 +1,31 @@
+package com.zoolatech.lecture5.tasks._4;
+
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+/**
+ * Create a method that accepts a string, counts all unique non-space character
+ * occurrences and displays each character with an amount of occurrences (output
+ * should be sorted by a character value in ascending order). Also, add a couple
+ * of examples that use your method.
+ */
+
+public class Task4 {
+    public static void main(String[] args) {
+        countCharacters("Hello World");
+    }
+
+    public static void countCharacters(String line) {
+        SortedMap<Character, Integer> sortedMap = new TreeMap<>();
+
+        for (char ch : line.toCharArray()) {
+            if (!sortedMap.containsKey(ch) && ch != ' ') {
```

Comment:
1. `ch != ' '` is duplicated, I think you could use smth like this:
```
if(ch != ' ') {
    if(map.contains...){
        ...
    } else {
        ...
    }
}
```
2. Map has a method `getOrDefault) that would allow you to remove the if condition